components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/client/IdentityProviderMgtServiceClient.java

```diff
@@ -113,20 +122,115 @@ public void updateResidentIdP(IdentityProvider identityProvider) throws Exceptio
+     * @throws IdentityException Error when getting list of Identity Providers for a matching filter with pagination.
+     */
+    public IdentityProvider[] getPaginatedIdpInfo(int pageNumber, String filter) throws IdentityException {
+
+        try {
+            return idPMgtStub.getPaginatedIdpInfo(filter, pageNumber);
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
+            log.error( "Error in retrieving the list of filtered Identity Providers for a given tenant", e);
+            throw new IdentityException(
+                    "Error in retrieving the list of filtered Identity Providers for a given tenant");
+        }
+    }
+
+    /**
+     * Get count of all Identity providers.
+     *
+     * @return total number of Idps.
+     * @throws IdentityException Error when getting count of total Identity Providers.
+     */
+    public int getIdpCount() throws IdentityException {
+
+        try {
+            return idPMgtStub.getAllIdpCount();
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
+            log.error("Error in retrieving the count of Identity Providers for a given tenant", e);
+            throw new IdentityException("Error in retrieving the count of Identity Providers for a given tenant");
+        }
+    }
+
+    /**
+     * Get count of all basic applications for a matching filter.
+     *
+     * @param filter Idp filter
+     * @return Count of applications match the filter
+     * @throws IdentityException Error when getting count of filtered total Identity Providers.
+     */
+    public int getFilteredIdpCount(String filter) throws IdentityException {
+
+        try {
+            return idPMgtStub.getFilteredIdpCount(filter);
+        } catch (RemoteException | IdentityProviderMgtServiceIdentityProviderManagementExceptionException e) {
+            log.error("Error in retrieving the count of registered Identity Providers for a given tenant", e);
+            throw new IdentityException(
+                    "Error in retrieving the count of registered Identity Providers for a given tenant");
+        }
+    }
+
+    /**
+     * Returns if pagination is supported in the remote API.
+     * This API is introduced to ensure that nothing breaks by invoking paginated APIs when API backend is not
+     * available.
+     *
+     * @return 'true' if pagination is supported
+     */
+    public boolean isPaginationSupported() {
+
+        if (!paginationSupportVerified) {
+            try {
+                this.getIdpCount();
+                paginationSupported = true;
+                if (log.isDebugEnabled()) {
+                    log.debug("IdentityApplicationManagementService stub and backend supports for SP pagination. Set " +
+                            "'paginationSupported' flag to 'true'.");
+                }
+            } catch (IdentityException | NoSuchMethodError e) {
```

Comment:
Is this both exception throws for the same reason? or else shall we use separate catch blocks and handle the separate exceptions and please check this throughout the PR.